src/main/java/org/opentripplanner/util/TranslationHelper.java

```diff
@@ -0,0 +1,79 @@
+package org.opentripplanner.util;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.onebusaway.gtfs.model.Translation;
+
+public final class TranslationHelper {
+
+    private static final String TABLE_NAME_FEED_INFO = "feed_info";
+    private static String feedLanguage = null;
+
+    private static Map<String, Map<Optional<String>, List<Translation>>> translationMap =
+            new HashMap<>();
+
+    public TranslationHelper(Collection<Translation> allTranslations, String language) {
+        feedLanguage = language;
+        Map<String, List<Translation>> byTableName =
+                allTranslations.stream().collect(Collectors.groupingBy(t -> t.getTableName()));
+        for (String tableName : byTableName.keySet()) {
+            if (tableName.equals(TABLE_NAME_FEED_INFO)) {
+                translationMap.put(tableName, byTableName.get(tableName)
+                        .stream()
+                        .collect(Collectors.groupingBy(t -> Optional.ofNullable(""))));
+            }
+            else {
+                translationMap.put(tableName + "_by_record", byTableName.get(tableName)
+                        .stream()
+                        .filter(t -> t.getFieldValue() == null)
+                        .collect(Collectors.groupingBy(t -> t.getRecordSubId() != null
+                                ? Optional.ofNullable(
+                                String.join("_", t.getRecordId(), t.getRecordSubId()))
```

Comment:
Add a comment about how the data structure looks like for the different cases

Reply:
Added.